lib/services/arweave/arweave_service.dart

```diff
@@ -329,6 +329,45 @@ class ArweaveService {
     }
   }
 
+  /// Gets the drive privacy of the latest drive entity with the provided id.
+  ///
+  /// This function first checks for the owner of the first instance of the [DriveEntity]
+  /// with the specified id and then queries for the latest instance of the [DriveEntity]
+  /// by that owner.
+  ///
+  /// Returns `null` if no valid drive is found.
+  Future<String?> getDrivePrivacyForId(String driveId) async {
+    final firstOwnerQuery = await _gql.execute(FirstDriveEntityWithIdOwnerQuery(
+        variables: FirstDriveEntityWithIdOwnerArguments(driveId: driveId)));
+
+    if (firstOwnerQuery.data!.transactions.edges.isEmpty) {
+      return null;
+    }
+
+    final driveOwner =
+        firstOwnerQuery.data!.transactions.edges.first.node.owner.address;
+
+    final latestDriveQuery = await _gql.execute(LatestDriveEntityWithIdQuery(
+        variables: LatestDriveEntityWithIdArguments(
+            driveId: driveId, owner: driveOwner)));
+
+    final queryEdges = latestDriveQuery.data!.transactions.edges;
+    if (queryEdges.isEmpty) {
+      return null;
+    }
+
+    final driveTx = queryEdges.first.node;
+
+    switch (driveTx.getTag(EntityTag.drivePrivacy)) {
+      case DrivePrivacy.public:
+        return DrivePrivacy.public;
+      case DrivePrivacy.private:
+        return DrivePrivacy.private;
+      default:
+        return null;
+    }
```

Comment:
It's redundant. Does it make sense to make this? 
```suggestion
   final privacy = driveTx.getTag(EntityTag.drivePrivacy);
    switch (privacy) {
      case DrivePrivacy.public:
      case DrivePrivacy.private:
        return privacy;
      default:
        return null;
    }
```

Reply:
[Updated](https://github.com/ardriveapp/ardrive-web/pull/376/commits/3879074f9dfead66893c50d396d4d11464693729). I don't know why I made it unnecessarily complex, probably a leftover from something I tried 😅 .